page-tracking/src/page_info.rs

```diff
@@ -134,29 +142,53 @@ impl PageInfo {
         self.mem_type
     }
 
+    /// Releases an exclusive reference to the page.
+    fn unlock_page(&mut self) -> PageTrackingResult<()> {
```

Comment:
Similar to the previous `lock()` vs `lock_page()` comment, I think you can get rid of `unlock_page()` entirely.